File: ruby/lib/svix/api/endpoint_transformation.rb
# frozen_string_literal: true
# This file is @generated

require "net/http"

module Svix
  class EndpointTransformation
    def initialize(client)
      @client = client
    end

    def get(app_id, endpoint_id)
      res = @client.execute_request(
        "GET",
        "/api/v1/app/#{app_id}/endpoint/#{endpoint_id}/transformation"
      )
      EndpointTransformationOut.deserialize(res)
    end

    def patch(app_id, endpoint_id, endpoint_transformation_patch)
      @client.execute_request(
        "PATCH",
        "/api/v1/app/#{app_id}/endpoint/#{endpoint_id}/transformation",
        body: endpoint_transformation_patch
      )
    end

  end
end

File: php/src/Api/EndpointTransformation.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Api;

use Svix\Exception\ApiException;
use Svix\Models\EndpointTransformationOut;
use Svix\Models\EndpointTransformationPatch;
use Svix\Request\SvixHttpClient;

class EndpointTransformation
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /**
     * Get the transformation code associated with this endpoint.
     *
     * @throws ApiException
     */
    public function get(
        string $appId,
        string $endpointId,
    ): EndpointTransformationOut {
        $request = $this->client->newReq('GET', "/api/v1/app/{$appId}/endpoint/{$endpointId}/transformation");
        $res = $this->client->send($request);

        return EndpointTransformationOut::fromJson($res);
    }

    /**
     * Set or unset the transformation code associated with this endpoint.
     *
     * @throws ApiException
     */
    public function patch(
        string $appId,
        string $endpointId,
        EndpointTransformationPatch $endpointTransformationPatch,
    ): void {
        $request = $this->client->newReq('PATCH', "/api/v1/app/{$appId}/endpoint/{$endpointId}/transformation");
        $request->setBody(json_encode($endpointTransformationPatch));
        $res = $this->client->sendNoResponseBody($request);
    }
}
